src/pages/posts/use-github-actions-for-ci.md

```diff
@@ -0,0 +1,227 @@
+---
+title: Start using GitHub Actions for CI
+date: '2019-03-07'
+spoiler: GitHub Actions can do a lot, including CI. Let's look at how to do it right!
+---
+
+GitHub Actions can do a whole lot of things - [delete branches](https://github.com/jessfraz/branch-cleanup-action), [compose tweets via pull requests](https://github.com/gr2m/twitter-together); it's kind of a general "thing do-er." However, when people hear about Actions, they often ask "is GitHub Actions a replacement for CI providers?" It's a great platform to have CI built into your workflow with minimal setup but a lot of control, so its a totally reasonable question.
+
+I get to use everyones' favorite answer: **it depends**! In this post, I'll share a workflow that I've been using for my Node.js projects, as well as some extensions to it for additional functionality. We'll also take a look at features of popular CI tools and see how they map to GitHub Actions.
+
+## Some pre-reading
+
+I'll be delving into the nitty gritty of writing a workflow file, including some lesser-known functionality. I'd encourage you to read [my previous blog post on GitHub Actions workflows](./what-are-github-workflows), where I talk a bit about workflows as a whole.
+
+You may also want to familiarize yourself with the [actions/bin repo](https://github.com/actions/bin), a collection of actions that are highly scoped and useful for composing a workflow without writing any custom code (especially [actions/bin/filter](https://github.com/actions/bin/blob/master/filter)).
+
+## But I love `{{ ci_provider }}` - why should I care?
+
+You'll see the benefits soon, but I'm not saying that existing projects migrate their CI to actions; rather, new projects benefit from the minimal setup of a CI workflow if you know what you're doing. Are GitHub Actions the _best_ CI platform? I'd say it really depends on your needs; if you're just running test, you can get a lot of functionality with very little effort (I know, the dream :heart_eyes:).
+
+
+## A typical Node.js workflow
+
+This is a `main.workflow` file that I've been using for a couple of projects and it's been a great experience. With most CI providers, I need to follow these steps:
+
+1. Push a configuration file to my repo
+2. Go to their site and find my repo
+3. Hit a checkbox/toggle to enable that CI provider to build my repo
+4. Push again to trigger CI
+
+Nowadays, I create my `.github/main.workflow` file and this is my whole process:
+
+1. Push a configuration file to my repo
+
+75% efficiency improvement! Because most of my projects follow the exact same CI patterns and have the exact same requirements, it really is repetitive. So, here's the `main.workflow` file in it's entirety:
```

Comment:
```suggestion
75% efficiency improvement! Because most of my projects follow the exact same CI patterns and have the exact same requirements, it really is repetitive. So, here's the `main.workflow` file in its entirety:
```